src/vs/vscode.d.ts

```diff
@@ -3429,6 +3460,15 @@ declare namespace vscode {
 		 * @return A new status bar item.
 		 */
 		export function createStatusBarItem(alignment?: StatusBarAlignment, priority?: number): StatusBarItem;
+
+		/**
+		 * Creates a [Terminal](#Terminal).
+		 *
+		 * @param name The optional name of a terminal, this will override the label used in the
+		 * terminal dropdown.
+		 * @return A new Terminal.
+		 */
+		export function createTerminal(name?: string): Thenable<vscode.Terminal>;
```

Comment:
`vscode.window.showInputBox` returns a `Thenable` though? The promise isn't due to the fact that it's executing in another process but that creating a terminal will show and focus it and potentially create the terminal panel before creating the terminal. While the first points can be worked around, I don't think the latter can be easily since the TerminalPanel may not be initialized when the API is called into?